rpc/errors.go

```diff
@@ -117,3 +117,10 @@ type internalServerError struct {
 func (e *internalServerError) ErrorCode() int { return e.code }
 
 func (e *internalServerError) Error() string { return e.message }
+
+// timeout reached
+type timeoutError struct{}
+
+func (e *timeoutError) ErrorCode() int { return 408 }
+
+func (e *timeoutError) Error() string { return "request timed out" }
```

Comment:
The error code needs to be in the `-32xxx` range (it's an "implementation-defined server error").